### PR TITLE
encoding: bazelify stringer

### PR DESCRIPTION
**Pull request by otan**
This one is a total hack. See comment for details.

Release note: None

### CODE REVIEW

File: pkg/util/encoding/BUILD.bazel [hunk 1]
```diff
@@ -60,33 +60,36 @@ go_test(
...
-# TODO (alanmas): Solve stringer issue "stringer: can't handle non-integer constant type Type"
-# Seems that we need to include pkg/util/encoding/encodingtype somehow.
-# We already tried to copy the source files over so bazel source now are enconding and encodingtype
-# but it is still failing due to the same error.
+genrule(
+    name = "gen-type-stringer",
+    srcs = [
+        "encoding.go",
+    ],
+    outs = ["type_string.go"],
```
Comment (alan-mas) [inline pkg/util/encoding/BUILD.bazel:69]: I think this empty space is not needed it.

**Review by alan-mas — APPROVED** [review]
Finally is working! you totally rocks!

LGTM

**Comment by otan** [comment]
bors r=alan-mas

**Comment by irfansharif** [comment]
🤮

**Pull request merged**